### PR TITLE
S185270: verify audience field in SAMLResponse

### PR DESCRIPTION
**Pull request by kworam-rally**

### CODE REVIEW

File: src/test/java/com/rallydev/saml/SAMLResponseValidatorTest.java [hunk 1]
```diff
@@ -202,26 +226,27 @@ public void supportAdditionalAttributes() throws IOException, SamlException, Val
...
+        SAMLResponseValidator validator = SAMLUtils.createSAMLResponseValidator(metdata.getBytes(StandardCharsets.UTF_8), MockSAMLBuilder.DEFAULT_AUDIENCE);
         Response response = validator.readAndValidateSAMLResponse(responseString);
 
 
         Map<String, String> parsedAttributes = SAMLUtils.getAttributes(response);
         System.out.println(parsedAttributes);
-        assertEquals(parsedAttributes.size(), numDefaultAttributes + 1);
+        assertEquals(parsedAttributes.size(), numAttributes);
         assertEquals(parsedAttributes, attributes);
     }
```
Comment (nedgladstone-rally) [inline src/test/java/com/rallydev/saml/SAMLResponseValidatorTest.java]: Need a bad audience test?

Comment (kworam-rally) [reply src/test/java/com/rallydev/saml/SAMLResponseValidatorTest.java]: done

Comment (kworam-rally) [reply src/test/java/com/rallydev/saml/SAMLResponseValidatorTest.java]: done

File: src/test/java/com/rallydev/saml/SAMLResponseValidatorTest.java [hunk 2]
```diff
@@ -33,66 +33,111 @@
...
+        System.out.println(parsedAttributes);
+        Map<String, String> expectedAttributes = getDefaultAttributes();
+        expectedAttributes.put(SAMLResponseValidator.AUDIENCE_REQUIRED_SAML_RESPONSE_CONDITION, audience);
+
+        assertEquals(parsedAttributes.size(), expectedAttributes.size());
+        assertEquals(parsedAttributes, expectedAttributes);
+    }
 
-        SAMLResponseValidator validator = SAMLUtils.createSAMLResponseValidator(defaultMetadata.getBytes(StandardCharsets.UTF_8));
+    @Test(expectedExceptions = ValidationException.class, expectedExceptionsMessageRegExp = "the SAML response does not have the expected audience: expected=bogusAudience  actual=https://rally1.rallydev.com")
```
[the remark this answers is not in the document]
Comment (kworam-rally) [reply src/test/java/com/rallydev/saml/SAMLResponseValidatorTest.java]: no.  The response contains a bogus audience when we expect the default audience.

[the remark this answers is not in the document]
Comment (nedgladstone-rally) [reply src/test/java/com/rallydev/saml/SAMLResponseValidatorTest.java]: But your comment says, "expected=bogusAudience  actual=https://rally1.rallydev.com". That's backward.

**Pull request merged**